nixos/lib/make-ext4-fs.nix

```diff
@@ -42,38 +48,39 @@ pkgs.stdenv.mkDerivation {
       bytes=$((2 * 4096 * $numInodes + 4096 * $numDataBlocks))
       echo "Creating an EXT4 image of $bytes bytes (numInodes=$numInodes, numDataBlocks=$numDataBlocks)"
 
-      truncate -s $bytes $out
-      faketime -f "1970-01-01 00:00:01" mkfs.ext4 -L ${volumeLabel} -U ${uuid} $out
+      truncate -s $bytes ./$img
+      faketime -f "1970-01-01 00:00:01" mkfs.ext4 -L ${volumeLabel} -U ${uuid} ./$img
 
       # Also include a manifest of the closures in a format suitable for nix-store --load-db.
       cp ${sdClosureInfo}/registration nix-path-registration
-      cptofs -t ext4 -i $out nix-path-registration /
+      cptofs -t ext4 -i ./$img nix-path-registration /
 
       # Create nix/store before copying paths
       faketime -f "1970-01-01 00:00:01" mkdir -p nix/store
-      cptofs -t ext4 -i $out nix /
+      cptofs -t ext4 -i ./$img nix /
 
       echo "copying store paths to image..."
-      cptofs -t ext4 -i $out $storePaths /nix/store/
+      cptofs -t ext4 -i ./$img $storePaths /nix/store/
 
       (
       echo "copying files to image..."
-      cd ./files
-      cptofs -t ext4 -i $out ./* /
+      pushd ./files
+      cptofs -t ext4 -i ../$img ./* /
+      popd
```

Comment:
Revert to `cd ./files`; the subshell is safer than `pushd/popd` here. The subshell (marked by the parentheses) does not change the PWD outside of the parentheses.

This change is basically a no-op adding a line to the code :).